ext/scheduler/airflow2/airflow_test.go

```diff
@@ -267,4 +268,182 @@ func TestAirflow2(t *testing.T) {
+
+			assert.Nil(t, err)
+			assert.Equal(t, expectedStatus, status)
+		})
+		t.Run("should return all dag run status when total entries is greater than page limit request", func(t *testing.T) {
+			respStringFirst := `{
+    "dag_runs": [
+        {
+            "conf": {},
+            "dag_id": "sample_select",
+            "dag_run_id": "scheduled__2020-03-25T02:00:00+00:00",
+            "end_date": "2020-06-01T17:32:58.489042+00:00",
+            "execution_date": "2020-03-25T02:00:00+00:00",
+            "external_trigger": false,
+            "start_date": "2020-06-01T16:32:58.489042+00:00",
+            "state": "success"
+        },
+        {
+            "conf": {},
+            "dag_id": "sample_select",
+            "dag_run_id": "scheduled__2020-03-26T02:00:00+00:00",
+            "end_date": "2020-06-01T16:33:01.020645+00:00",
+            "execution_date": "2020-03-26T02:00:00+00:00",
+            "external_trigger": false,
+            "start_date": "2020-06-01T16:33:01.020645+00:00",
+            "state": "failed"
+        }
+    ],
+    "total_entries": 3
+}`
+			respStringSecond := `{
+    "dag_runs": [
+        {
+            "conf": {},
+            "dag_id": "sample_select",
+            "dag_run_id": "scheduled__2020-03-27T02:00:00+00:00",
+            "end_date": "2020-06-01T16:34:01.020645+00:00",
+            "execution_date": "2020-03-26T02:00:00+00:00",
+            "external_trigger": false,
+            "start_date": "2020-06-01T16:35:01.020645+00:00",
+            "state": "running"
+        }
+    ],
+    "total_entries": 3
+}`
+			expectedExecutionTime0, _ := time.Parse(models.InstanceScheduledAtTimeLayout, "2020-03-25T02:00:00+00:00")
+			expectedExecutionTime1, _ := time.Parse(models.InstanceScheduledAtTimeLayout, "2020-03-26T02:00:00+00:00")
+			expectedExecutionTime2, _ := time.Parse(models.InstanceScheduledAtTimeLayout, "2020-03-26T02:00:00+00:00")
+			expectedStatus := []models.JobStatus{
+				{
+					ScheduledAt: expectedExecutionTime0,
+					State:       models.JobStatusStateSuccess,
+				},
+				{
+					ScheduledAt: expectedExecutionTime1,
+					State:       models.JobStatusStateFailed,
+				},
+				{
+					ScheduledAt: expectedExecutionTime2,
+					State:       models.JobStatusStateRunning,
+				},
+			}
+
+			dagRunResp := []io.ReadCloser{
+				ioutil.NopCloser(bytes.NewReader([]byte(respStringFirst))),
+				ioutil.NopCloser(bytes.NewReader([]byte(respStringSecond))),
+			}
+			countDagRunReq := -1
+			client := &MockHttpClient{
+				DoFunc: func(req *http.Request) (*http.Response, error) {
+					countDagRunReq++
+					return &http.Response{
+						StatusCode: http.StatusOK,
+						Body:       dagRunResp[countDagRunReq],
+					}, nil
+				},
+			}
+
+			air := airflow2.NewScheduler(nil, client)
+			status, err := air.GetDagRunStatus(ctx, projectSpec, jobName, startDateTime, endDateTime, batchSize)
+
+			assert.Nil(t, err)
+			assert.Equal(t, expectedStatus, status)
```

Comment:
added status value assertion and added value/error assert to other test cases in this file too.